### PR TITLE
Feature/copy json rpc files to android core

### PR DESCRIPTION
**Pull request by jakobuid**

### CODE REVIEW

**Review by TalhaAli00 — APPROVED** [review]

File: android_core/src/main/kotlin/com/walletconnect/android_core/common/model/vo/sync/PendingRequestVO.kt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package com.walletconnect.sign.core.model.vo.sync
+package com.walletconnect.android_core.common.model.vo.sync
 
 data class PendingRequestVO(
```
Comment (Elyniss) [inline android_core/src/main/kotlin/com/walletconnect/android_core/common/model/vo/sync/PendingRequestVO.kt:3]: Should we remove `VO` suffix?

File: android_core/src/main/kotlin/com/walletconnect/android_core/di/JsonRpcModule.kt [hunk 2]
```diff
@@ -1,18 +1,11 @@
...
+import com.walletconnect.android_core.json_rpc.data.NetworkState
 import org.koin.dsl.module
 
 fun jsonRpcModule() = module {
-    single {
-        JsonRpcSerializer(get())
-    }
 
     single {
         NetworkState(get())
```
Comment (Elyniss) [inline android_core/src/main/kotlin/com/walletconnect/android_core/di/JsonRpcModule.kt:9]: maybe we can move it `NetworkModule`?

File: android_core/src/main/kotlin/com/walletconnect/android_core/json_rpc/data/JsonRpcSerializerAbstract.kt [hunk 3]
```diff
@@ -0,0 +1,15 @@
+@file:JvmSynthetic
+
+package com.walletconnect.android_core.json_rpc.data
+
+import com.squareup.moshi.Moshi
+import com.walletconnect.android_core.common.model.type.ClientParams
+import com.walletconnect.android_core.common.model.type.SerializableJsonRpc
+
+abstract class JsonRpcSerializerAbstract(open val moshi: Moshi) {
```
Comment (Elyniss) [inline android_core/src/main/kotlin/com/walletconnect/android_core/json_rpc/data/JsonRpcSerializerAbstract.kt:9]: mmm, love it

File: android_core/src/main/kotlin/com/walletconnect/android_core/utils/UtilFunctions.kt [hunk 4]
```diff
@@ -0,0 +1,5 @@
+@file:Suppress("PackageDirectoryMismatch")
+
+package com.walletconnect.utils
```
Comment (Elyniss) [inline android_core/src/main/kotlin/com/walletconnect/android_core/utils/UtilFunctions.kt:3]: nice

File: foundation/src/main/kotlin/com/walletconnect/foundation/util/UtilFunctions.kt [hunk 5]
```diff
@@ -5,15 +5,12 @@ package com.walletconnect.util
 import java.security.SecureRandom
 import javax.ws.rs.core.UriBuilder
 
-@JvmSynthetic
```
Comment (Elyniss) [inline foundation/src/main/kotlin/com/walletconnect/foundation/util/UtilFunctions.kt:8]: Nice catch

File: android_core/src/main/kotlin/com/walletconnect/android_core/common/model/vo/json_rpc/JsonRpcResponse.kt [hunk 6]
```diff
@@ -1,27 +1,27 @@
-package com.walletconnect.sign.core.model.vo.jsonRpc
+package com.walletconnect.android_core.common.model.vo.json_rpc
```
Comment (Elyniss) [inline android_core/src/main/kotlin/com/walletconnect/android_core/common/model/vo/json_rpc/JsonRpcResponse.kt:1]: Should we remove `.vo` level from package hierarchy?

File: android_core/src/main/kotlin/com/walletconnect/android_core/common/model/vo/sync/PendingRequestVO.kt [hunk 7]
```diff
@@ -1,4 +1,4 @@
-package com.walletconnect.sign.core.model.vo.sync
+package com.walletconnect.android_core.common.model.vo.sync
```
Comment (Elyniss) [inline android_core/src/main/kotlin/com/walletconnect/android_core/common/model/vo/sync/PendingRequestVO.kt:1]: Should we remove `.vo` level from package hierarchy?

File: android_core/src/main/kotlin/com/walletconnect/android_core/common/model/vo/sync/WCRequest.kt [hunk 8]
```diff
@@ -1,9 +1,9 @@
-package com.walletconnect.sign.core.model.vo.sync
+package com.walletconnect.android_core.common.model.vo.sync
```
Comment (Elyniss) [inline android_core/src/main/kotlin/com/walletconnect/android_core/common/model/vo/sync/WCRequest.kt:1]: Should we remove `.vo` level from package hierarchy?

File: android_core/src/main/kotlin/com/walletconnect/android_core/common/model/vo/sync/WCResponse.kt [hunk 9]
```diff
@@ -1,12 +1,12 @@
-package com.walletconnect.sign.core.model.vo.sync
+package com.walletconnect.android_core.common.model.vo.sync
```
Comment (Elyniss) [inline android_core/src/main/kotlin/com/walletconnect/android_core/common/model/vo/sync/WCResponse.kt:1]: Should we remove `.vo` level from package hierarchy?

File: signSDK/sign/src/main/kotlin/com/walletconnect/sign/client/mapper/ClientMapper.kt [hunk 10]
```diff
@@ -17,7 +17,7 @@ internal fun EngineDO.ProposedSequence.toClientProposedSequence(): Sign.Model.Pr
     }
 
 @JvmSynthetic
-internal fun Sign.Model.JsonRpcResponse.toJsonRpcResponseVO(): JsonRpcResponseVO =
+internal fun Sign.Model.JsonRpcResponse.toJsonRpcResponseVO(): JsonRpcResponse =
```
Comment (Elyniss) [inline signSDK/sign/src/main/kotlin/com/walletconnect/sign/client/mapper/ClientMapper.kt:20]: Naming confusing

File: signSDK/sign/src/main/kotlin/com/walletconnect/sign/di/JsonRpcModule.kt [hunk 11]
```diff
@@ -0,0 +1,19 @@
+package com.walletconnect.sign.di
+
+import com.walletconnect.sign.json_rpc.data.JsonRpcSerializer
+import com.walletconnect.sign.json_rpc.domain.JsonRpcInteractor
+import org.koin.dsl.module
+import com.walletconnect.android_core.di.jsonRpcModule as androidCoreJsonRcpModule
+
+fun jsonRpcModule() = module {
+
+    includes(androidCoreJsonRcpModule())
```
Comment (Elyniss) [inline signSDK/sign/src/main/kotlin/com/walletconnect/sign/di/JsonRpcModule.kt:10]: if we move NetworkState to NetworkModule this is not necessary

File: signSDK/sign/src/main/kotlin/com/walletconnect/sign/json_rpc/data/JsonRpcSerializer.kt [hunk 12]
```diff
@@ -39,11 +40,8 @@ internal class JsonRpcSerializer(private val moshi: Moshi) {
             is SessionRpcVO.SessionRequest -> trySerialize(payload)
             is SessionRpcVO.SessionDelete -> trySerialize(payload)
             is SessionRpcVO.SessionSettle -> trySerialize(payload)
-            is RelayerDO.JsonRpcResponse.JsonRpcResult -> trySerialize(payload)
-            is RelayerDO.JsonRpcResponse.JsonRpcError -> trySerialize(payload)
+            is JsonRpc.JsonRpcResponse.JsonRpcResult -> trySerialize(payload)
+            is JsonRpc.JsonRpcResponse.JsonRpcError -> trySerialize(payload)
             else -> String.Empty
```
Comment (Elyniss) [inline signSDK/sign/src/main/kotlin/com/walletconnect/sign/json_rpc/data/JsonRpcSerializer.kt:45]: We could add it to `JsonRpcSerializerAbstract` so we won't need to add it to every API

**Review by TalhaAli00 — APPROVED** [review]

**Pull request merged**